Include/internal/pycore_interp.h

```diff
@@ -230,6 +235,27 @@ PyAPI_FUNC(int) _PyInterpreterState_IDInitref(PyInterpreterState *);
 PyAPI_FUNC(int) _PyInterpreterState_IDIncref(PyInterpreterState *);
 PyAPI_FUNC(void) _PyInterpreterState_IDDecref(PyInterpreterState *);
 
+#define FREELIST_QUANTUM (2*sizeof(void*))
+#define SIZE_TO_FREELIST_INDEX(size) ((size-4)/FREELIST_QUANTUM)
+#define FREELIST_INDEX_TO_SIZE(idx) (FREELIST_QUANTUM*(idx) + 4)
```

Comment:
This must be a multiple of `FREELIST_QUANTUM`:

If `idx = (size + FREELIST_QUANTUM -1)/FREELIST_QUANTUM` then `size = idx*FREELIST_QUANTUM`

Reply:
This doesn't matter yet. We only have one size. 

What we should talk about now is (1) it crashes (2) I was unable to replicate the 1% speedup you saw on your branch: https://github.com/faster-cpython/benchmarking/tree/main/results/bm-20221014-3.12.0a0-d09385f

Reply:
^--- those are benchmark results from your branch.

Reply:
> If `idx = (size + FREELIST_QUANTUM -1)/FREELIST_QUANTUM` then `size = idx*FREELIST_QUANTUM`

`idx = ((size-4)/FREELIST_QUANTUM)` --> 
`idx * FREELIST_QUANTUM = size-4` -->
`idx * FREELIST_QUANTUM + 4 = size`

Reply:
For real numbers, yes. Not for integers.

Reply:
We need the following to be true:
`FREELIST_INDEX_TO_SIZE(idx) % FREELIST_QUANTUM == 0` for all `idx`
`FREELIST_INDEX_TO_SIZE(SIZE_TO_FREELIST_INDEX(size)) >= size` for all `size`

Which is not the case here.
Take FREELIST_QUANTUM = 16, then for size 17 we have:
`FREELIST_INDEX_TO_SIZE(SIZE_TO_FREELIST_INDEX(17)) == 4`

It might be clearer to rename `FREELIST_INDEX_TO_SIZE` to `FREELIST_INDEX_TO_ALLOCATED_SIZE` to indicate that it is the amount of memory allocated, not necessarily requested.